_search-plugins/search-shard-routing.md

```diff
@@ -0,0 +1,177 @@
+
+- The time it took particular node to run previous requests.
+- Latency between coordinating node and selected node.
+- The size of the queue of node's search thread pool.
+
+You can turn off this feature using following curl command with user which has permissions to call REST APIs, mode details on REST API can be found at [REST management API settings]({{site.url}}{{site.baseurl}}/install-and-configure/configuring-opensearch/security-settings/#rest-management-api-settings):
+
+```
+PUT /_cluster/settings
+{
+  "persistent": {
+    "cluster.routing.use_adaptive_replica_selection": false
+  }
+}
+```
+
+In this case, OpenSearch routing will use _round-robin_ which can impact the search latency negatively.
+{: .note}
+
+## Node and shard selection during searches
+
+OpenSearch uses all nodes to choose best routing for search requests, however, there might be reasons why you want to manually select which nodes or shards the search request is sent to, possible reasons could be:
+
+- Using caching of previous searches.
+- Dedicating specific hardware for searches.
+- Using only local nodes for searches.
+
+You can use the `preference` parameter in the search query to indicate the destination for the search. The following is the complete list of available options:
+
+1. `_primary`: Forces the search to execute only on primary shards.
+```
+GET /my-index/_search?preference=_primary
+```
+2. `_primary_first`: Prefers primary shards but will use replica shards if the primary is not available.
+```
+GET /my-index/_search?preference=_primary_first
+```
+3. `_replica`: Forces the search to execute only on replica shards.
+```
+GET /my-index/_search?preference=_replica
+```
+4. `_replica_first`: Prefers replica shards but will use primary shards if no replicas are available.
+```
+GET /my-index/_search?preference=_replica_first
+```
+5. `_only_nodes:<node-id>,<node-id>`: Limits the search to execute only on specific nodes by their IDs.
+```
+GET /my-index/_search?preference=_only_nodes:node-1,node-2
+```
+6. `_prefer_nodes:<node-id>,<node-id>`: Prefers to execute the search on specific nodes but will use other nodes if the preferred ones are not available.
+```
+GET /my-index/_search?preference=_prefer_nodes:node-1,node-2
+```
+7. `_shards:<shard-id>,<shard-id>`: Limits the search to specific shards.
+```
+GET /my-index/_search?preference=_shards:0,1
+```
+8. `_local`: Executes the search on the local node if the data is available there, which can reduce latency.
+```
+GET /my-index/_search?preference=_local
+```
+9. Custom string: You can use any custom string as the preference value. This custom string ensures that requests with the same string are routed to the same shards consistently, which can be useful for caching.
+```
+GET /my-index/_search?preference=custom_string
+```
+
+## Custom routing during index and search
+
+You can specify routing during both indexing and searching operations.
+
+### Routing during indexing
```

Comment:
should we also add risk associated with it as it introduces the risk of creating hotspots or data skew if the routing values are not evenly distributed across the shards.

Reply:
@rishabhmaurya thank you for the comments. Thats updated now